src/main/resources/project/server/stopDomain.pl

```diff
@@ -212,37 +216,39 @@ sub main {
         }
         else {
             my $error_summary = "Something wrong after stopping all slave host controllers (before stopping master host controller '$master_host').";
-            $error_summary .= "\nExpected is to have only master host controller '$master_host' started at this point, but actual list of started host controllers is: [" . join(", ", @all_hosts_after_all_slaves_shutdown) . "]";
+            $error_summary .= "\nExpected is to have only master host controller '$master_host' started at this point, but actual list of started host controllers is: ["
+                . join(", ", @all_hosts_after_all_slaves_shutdown) . "]";
             $jboss->log_error($error_summary);
 
             $summary .= "\n\nError: $error_summary";
-            $jboss->set_property(summary => $summary );
+            $jboss->set_property(summary => $summary);
             $jboss->error();
             exit 1;
         }
 
         # shutdown of master host controller
         $jboss->log_info("Starting shudown of master host controller '$master_host'");
-        my $cli_shutdown_master = "/host=$master_host:shutdown";
-        $cli_shutdown_master .= "(timeout=$param_timeout)" if $param_timeout ne "";
+        my $cli_shutdown_master = $jboss_is_6x
+            ? get_cli_host_shutdown_6x(host => $master_host)
+            : get_cli_host_shutdown_7x(host => $master_host, timeout => $param_timeout);
         run_command_with_exiting_on_error(command => $cli_shutdown_master, jboss => $jboss);
         $summary .= "\nShutdown was performed for master host controller '$master_host'";
         $jboss->log_info("Done with shudown of master host controller '$master_host'");
 
         # verification that shutdown of master host controller was successful
         $cli_command = ':read-attribute(name=launch-type)';
         my %result = $jboss->run_command($cli_command);
-        if ( $result{code}
+        if ($result{code}
             && ($result{stdout} =~ m/The\scontroller\sis\snot\savailable/gs
-                || $result{stderr} =~ m/The\scontroller\sis\snot\savailable/gs) ) {
+            || $result{stderr} =~ m/The\scontroller\sis\snot\savailable/gs)) {
```

Comment:
Function should be used in that case instead of complex conditions

Reply:
done